### PR TITLE
PR - Changeling Vox Transformation - Change to changeling_power.dm

### PR DESCRIPTION
**Pull request by Kiyahitayika**
## What Does This PR Do
This PR fixes Vox colors and gender in Changeling Powers for a more precise duplicate/clone of the Vox dna.

## Why It's Good For The Game
This allows Changelings to fully disguise themselves as Vox characters and/or transform others into proper "clones" of Vox characters. Reduces any meta-gaming from distinguishing an "original" Vox from a "transformed" Vox.

## Changelog
:cl:
fix: Vox coloration, gender when transforming self/others into Vox with Changeling Powers
/:cl:

### CODE REVIEW

File: code/game/gamemodes/changeling/changeling_power.dm [hunk 1]
```diff
@@ -101,4 +101,36 @@ the same goes for Remove(). if you override Remove(), call parent or else your p
...
+		if(577) //Vox Skintone 4, Grey Vox, Hex241 = 577 Log10
+			H.s_tone = 4
+		if(558) //Vox Skintone 5, Emerald Vox, Hex22E = 558 Log10
+			H.s_tone = 5
+		if(539) //Vox Skintone 6, Azure Vox, Hex21B = 539 Log10
+			H.s_tone = 6
+		else //Vox Skintone 0, Default Green Vox
+			H.s_tone = 0
+
+	H.change_gender(data["gender"]) //Get Vox Gender from Blood
```
[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/game/gamemodes/changeling/changeling_power.dm:121]: You know what, you're on point here. 

Short answer: forgot to test the changeling_powers after adding this one line.

I tested the toxins.dm one and absent-mindedly copy/pasted it before testing it in changeling_powers.dm. I was supposed to test this before submitting the changelings_powers.dm fix and work out how to get the gender. I'll work on this later today and figure it out and fix it up.

[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/game/gamemodes/changeling/changeling_power.dm:121]: Everything is also indented incorrectly. I'll fix it up! So sorry!

[the remark this answers is not in the document]
Comment (Kiyahitayika) [reply code/game/gamemodes/changeling/changeling_power.dm]: @SabreML helped so much on this! We had to get the gender from the mob the DNA was taken from.  The solution was to run a search based on the "real_name" from the DNA against the "real_name" of all "human" mobs. From the _match_, we then pull the gender of that mob and apply it to the transformed changeling/changeling target.

It's up to you and other admins/heads if you want to implement this. We're not clear if running this search will slow things down. If it does, I apologize. We can remove the "for" statement altogether. Gender won't match, but at least the colors will. Gender not matching for changelings changing into Vox is far from a game-breaking issue if it's removed.

**Review by farie82 — CHANGES_REQUESTED** [review]
Taking #14279 in consideration as well.
It is best to make this a proc on the species or such. So the species can respond to a change like this.
Try to extract the differences between a non vox and a vox and put that in a seperate proc on the vox species. So you'll have something like this:
```dm
/proc/thingy()
 species.on_dna_change(params)

/datum/species/proc/on_dna_change(params)
 do some stuff that goes for all species

/datum/species/vox/on_dna_change(params)
 ..() // either call it before or after the specific stuff. Depending on what needs to happen
 do specific stuff for vox.
```

It would be best to merge the two PRs I'd say since they both tackle the same issue and will have the same solution

**Comment by Kiyahitayika** [comment]
> Taking #14279 in consideration as well.
> It is best to make this a proc on the species or such. So the species can respond to a change like this.
> Try to extract the differences between a non vox and a vox and put that in a seperate proc on the vox species. So you'll have something like this:
> 
> ```dm
> /proc/thingy()
>  species.on_dna_change(params)
> 
> /datum/species/proc/on_dna_change(params)
>  do some stuff that goes for all species
> 
> /datum/species/vox/on_dna_change(params)
>  ..() // either call it before or after the specific stuff. Depending on what needs to happen
>  do specific stuff for vox.
> ```
> 
> It would be best to merge the two PRs I'd say since they both tackle the same issue and will have the same solution

Woah! This is awesome! I didn't know you could create an "on_dna_change()" change thingy! This is so cool. I am definitely going to look into this. Thank you so much for the advice. I absolutely agree that this would be a better approach.

This being my first PR and my first stab at coding in BYOND and for Paradise, I am definitely looking for feedback like this. I wasn't 100% happy with the solution I found and I really appreciate your input. Thank you!

**Comment by farie82** [comment]
> > Taking #14279 in consideration as well.
> > It is best to make this a proc on the species or such. So the species can respond to a change like this.
> > Try to extract the differences between a non vox and a vox and put that in a seperate proc on the vox species. So you'll have something like this:
> > ```dm
> > /proc/thingy()
> >  species.on_dna_change(params)
> > 
> > /datum/species/proc/on_dna_change(params)
> >  do some stuff that goes for all species
> > 
> > /datum/species/vox/on_dna_change(params)
> >  ..() // either call it before or after the specific stuff. Depending on what needs to happen
> >  do specific stuff for vox.
> > ```
> > 
> > 
> > It would be best to merge the two PRs I'd say since they both tackle the same issue and will have the same solution
> 
> Woah! This is awesome! I didn't know you could create an "on_dna_change()" proc call like that. This is so cool. I am definitely going to look into this. Thank you so much for the advice. I absolutely agree that this would be a better approach.
> 
> This being my first PR and my first stab at coding in BYOND and for Paradise, I am definitely looking for feedback like this. I wasn't 100% happy with the solution I found and I really appreciate your input. Thank you!

Feel free to PM me if you need some help. Or ask in coding chat!

**Comment by Kiyahitayika** [comment]
@farie82  There is...an alternative where I "fix" the DNA code to work with trinary-gendered species, Vox skintones, and Taj/Vulp tail styles. But that would take a longer time and require overhauling _too_ many files @__@

**Comment by Kiyahitayika** [comment]
> There is...an alternative where I "fix" the DNA code to work with trinary-gendered species, 
https://github.com/ParadiseSS13/Paradise/pull/14406

**Comment by Kiyahitayika** [comment]
Per @farie82 's suggestion and with @AffectedArc07 's permission, refractoring Vox to work with UpdateAppearance()

**Pull request closed without merging**